evennia/utils/evstring.py

```diff
@@ -0,0 +1,1497 @@
+        if self._ansi_render:
+            return self._ansi_render.convert_markup(self._code_chunks, xterm256=xterm256, rgb=rgb, mxp=mxp)
+        else:
+            # should we raise an error?
+            return self._clean_string
+
+    def to_html(self, renderer=None):
+        """
+        Returns a string object with Evennia markup converted to HTML tags
+        """
+        if self._html_render:
+            return self._html_render.convert_markup(self._renderable_chunks)
+        elif renderer:
+            try:
+                return renderer.convert_markup(self._renderable_chunks)
+            except:
+                # it was an invalid renderer
+                return self._clean_string
+        else:
+            # should we raise an error?
+            return self._clean_string
+
+    def partition(self, sep, reverse=False):
+        """
+        Splits once into three sections (with the separator being the middle section).
+        
+        Code segments of the EvString are not checked for the separator, as splitting
+        in the middle of one of those elements would break it. Links will use the
+        start of the text to a link and convert the tail to a normal EvString.
+
+        Args:
+            sep (str): The separator to split the string on.
+            reverse (boolean): Whether to split the string on the last
+                occurrence of the separator rather than the first.
+
+        Returns:
+            EvString: The part of the string before the separator
+            EvString: The separator itself
+            EvString: The part of the string after the separator.
+
+        """
+        result = tuple()
+        if reverse:
+            for i, chunk in reversed(enumerate(self._code_chunks)):
+                # we do not split within codes
+                if isinstance(chunk, EvCode):
+                    continue
+                result = chunk.rpartition(sep)
+                if result[0]:
+                    break
+                else:
+                    result = None
+        else:
+            # do the same thing, but forwards
+            for i, chunk in enumerate(self._code_chunks):
+                # we do not split within codes
+                if isinstance(chunk, EvCode):
+                    continue
+                result = chunk.partition(sep)
+                if result[-1]:
+                    break
+                else:
+                    result = None
+        
+        # now we check our results
+        if not result:
+            return ('', '', self) if reverse else (self, '', '')
+        
+        else:
+            # we broke the loop at i, so split the code chunks there
+            first = self._code_chunks[:i] + (result[0],)
+            last = (result[2],) + self._code_chunks[i+1:]
+            # create new EvStrings from our partitioned results and return
+            return ( EvString(''.join(first), chunks=first), sep, EvString(''.join(last), chunks=last), )
```

Comment:
Despite your own advice in `EvString.join()`'s doc string, you are joining EvStrings with `str.join()` here. ;)